.automation/copy-issue-labels-to-pr

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: Apache-2.0
+
+import itertools
+import json
+import re
+import sys
+from github import Github
+
+# We don't want to copy all labels on linked issues; only those in this subset.
+COPYABLE_LABELS = {
+    "amd sev",
+    "documentation",
+    "ibm pef",
+    "infrastructure",
+    "intel sgx",
+    "wasm",
+}
+
+REGEX = re.compile("(close[sd]?|fix|fixe[sd]?|resolve[sd]?)\s*:?\s+#(\d+)", re.I)
+
+# Get inputs from shell
+(token, repository, path) = sys.argv[1:4]
+
+# Authenticate with Github using our token
+g = Github(token)
+
+# Initialize repo
+repo = g.get_repo(repository)
+
+# Returns a pull request extracted from Github's event JSON.
+def get_pr(event):
+    # --- Extract PR from event JSON ---
+    # `check_run`/`check_suite` does not include any direct reference to the PR
+    # that triggered it in its event JSON. We have to extrapolate it using the head
+    # SHA that *is* there.
+
+    # Get head SHA from event JSON
+    pr_head_sha = event["check_suite"]["head_sha"]
+
+    # Find the repo PR that matches the head SHA we found
+    return {pr.head.sha: pr for pr in repo.get_pulls()}[pr_head_sha]
+
+# Get all issue numbers related to a PR.
+def get_related_issues(pr):
+    # Extract all associated issues from closing keyword in PR
+    for verb, num in REGEX.findall(pr.body):
+        yield int(num)
+
+    # Extract all associated issues from PR commit messages
+    results_commit_messages = [REGEX.findall(c.commit.message) for c in pr.get_commits()]
```

Comment:
Now that you have `yield`, I would ditch the list comprehension and itertools stuff here and just do a nested for loop.